lib/failure.sh

```diff
@@ -102,17 +102,43 @@ fail_iojs_unsupported() {
        }
 
        io.js merged back into Nodejs.org in 2015 and has been unsupported
-       for many years. It is likely to contain several large security 
+       for many years. It is likely to contain several large security
        vulnerabilities that have been patched in Node.
 
-       You can update your app to use the official Node.js release by 
+       You can update your app to use the official Node.js release by
        removing the version specfication under \"engines\" in your
        package.json.
        "
     fail
   fi
 }
 
+fail_yarn2_unsupported() {
+  local uses_yarn="$1"
+  local build_dir="$2"
+  local uses_yarn2
+
+  uses_yarn2=$(detect_yarn2 "$uses_yarn" "$build_dir")
+
+  if [[ "$uses_yarn2" == "true" ]]; then
+    mcount "failures.yarn2-unsupported"
+    meta_set "failure" "yarn2-unsupported"
```

Comment:
It will be interesting to see how many of these there are :)

Reply:
yep, agreed!